File: euler/py/euler.py
#!/usr/bin/python
# by protago90

from argparse import ArgumentParser, Namespace
from typing import Callable, List
from src import *  # i.e. all solvers from src/E*.py


class Solver():
    e1:  Callable = e001.solve_e1   # Multiples of 5 and 3    
    e2:  Callable = e002.solve_e2   # Even Fibonacci numbers
    e3:  Callable = e003.solve_e3   # Largest prime factor
    e4:  Callable = e004.solve_e4   # Largest polindrome product
    e5:  Callable = e005.solve_e5   # Smallest multiple
    e6:  Callable = e006.solve_e6   # Sum square difference
    e7:  Callable = e007.solve_e7   # 10001st
    e8:  Callable = e008.solve_e8   # Largest product in a series
    e9:  Callable = e009.solve_e9   # Special Pythagorean triplets
    e10: Callable = e010.solve_e10  # Summation of primes

    @classmethod
    def run(cls, n: str) -> None:
        return getattr(cls, n)()

    @classmethod
    def get_done(cls) -> List[int]:
        return [int(''.join(filter(str.isdigit, k))) for k, v in cls.__dict__.items()
                if not k.startswith('__') and callable(v)]


def parse_args() -> Namespace:
    s: List[int] = Solver.get_done()
    a = ArgumentParser()
    a.add_argument('-p', '--problem', required=True, type=int, choices=s)
    return a.parse_args()


if __name__ == '__main__':
    euler_problem: str = 'e{:01d}'.format(parse_args().problem)
    Solver.run(n=euler_problem)

File: euler/c/euler.c
// by protago90

#include <stdio.h>
#include <stdlib.h>
#include "utils.h"
#include "eall.h"  // i.e. all solvers from src/E??.c


void run_solver(int n) {
    long (*solve)();
    switch (n) {
        case 1:  solve = solve_e1;  break;   // Multiples of 5 and 3
        case 2:  solve = solve_e2;  break;   // Even Fibonacci numbers
        case 3:  solve = solve_e3;  break;   // Largest prime factor
        case 4:  solve = solve_e4;  break;   // Largest polindrome product
        case 5:  solve = solve_e5;  break;   // Smallest multiple
        case 6:  solve = solve_e6;  break;   // Sum square difference
        case 7:  solve = solve_e7;  break;   // 10001st
        case 8:  solve = solve_e8;  break;   // Largest product in a series	
        case 9:  solve = solve_e9;  break;   // Special Pythagorean triplets
        case 10: solve = solve_e10; break;   // Summation of primes  
        default: 
            printf("  >> given euler problem is not valid\n"); return;
    }
    promptify(solve, n);
}


int main (int argc, char *argv[]) {
    if (argc <= 1) {
        printf("  >> specify #n euler problem\n"); return 0;
    }
    int euler_problem = atoi(argv[1]); 
    run_solver(euler_problem);
    return 0;
}